mongorestore/mongorestore_test.go

```diff
@@ -1658,6 +1659,163 @@ func createTimeseries(dbName, coll string, client *mongo.Client) {
 	client.Database(dbName).RunCommand(context.Background(), createCmd)
 }
 
+func TestRestoreTimeseriesCollectionsWithMixedSchema(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	ctx := context.Background()
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	if err != nil {
+		t.Fatalf("No cluster available: %v", err)
+	}
+
+	defer sessionProvider.Close()
+
+	session, err := sessionProvider.GetSession()
+	if err != nil {
+		t.Fatalf("No client available")
+	}
+
+	fcv := testutil.GetFCV(session)
+	// TODO: Enable tests for 6.0, 7.0 and 8.0 (TOOLS-3597).
+	// The server fix for SERVER-84531 was only backported to 7.3.
+	if cmp, err := testutil.CompareFCV(fcv, "7.3"); cmp < 0 {
+		if err != nil {
+			t.Fatalf("Failed to get FCV: %v", err)
+		}
+		t.Skip("Requires server with FCV 7.3 or later")
+	}
+
+	if cmp, err := testutil.CompareFCV(fcv, "8.0"); cmp >= 0 {
```

Comment:
Let's just add a TODO here with a linked REP ticket that's blocked on that backport? Or if not a TODO, as long as we have the info in the ticket that's fine.

Reply:
I added that info to [TOOLS-3597](https://jira.mongodb.org/browse/TOOLS-3597).